htlcswitch/interfaces.go

```diff
@@ -530,3 +530,27 @@ type AuxTrafficShaper interface {
 	// meaning that it's from a custom channel.
 	IsCustomHTLC(htlcRecords lnwire.CustomRecords) bool
 }
+
+// AttemptStore defines the interface for storing and managing the results
+// of HTLC payment attempts. It is designed to support both local and remote
+// lifecycle controllers, allowing full control over result storage and cleanup.
```

Comment:
what do you mean by life-cycle controllers, lifecycle of what in particular ?

Reply:
The `ChannelRouter` has an aspect to it called `paymentLifecycle` so I've gotten to referring to the router as a "payment life-cycle controller" amongst it's other responsibilities like path-finding, onion construction, etc. I can update the comment to be a bit more clear here if you prefer.